### ISSUE TITLE
InstallationKernel redirects to wrong route

### ISSUE THREAD

**Issue by fritzmg** [contao/installation-bundle#18]
The InstallationKernel redirects to the indexAction of the InstallationController, whenever the system is able to boot:

``` php
if ($this->canBootRealSystem()) {
    return new RedirectResponse('../contao/install');
}
```

https://github.com/contao/installation-bundle/blob/1.1.0/src/HttpKernel/InstallationKernel.php#L70

However, this redirect does not work when you access the Install Tool like this:

```
http://dev.env/contao-4.2.0/web/contao/install.php
```

because it will then redirect to

```
http://dev.env/contao-4.2.0/contao/install
```

due to `../`

**Comment by leofeyer** [comment]
Looks ok to me. You are supposed to set the document root to the `web/` folder.

**Comment by fritzmg** [comment]
> Looks ok to me. You are supposed to set the document root to the web/ folder.

It's no a requirement to run Symfony applications though.

The current redirect is simply wrong. It only works in an environment where you set the document root to `/web` by "chance". For instance, if you open the Install Tool via

```
http://example.org/install.php
```

then

``` php
RedirectResponse('../contao/install');
```

wants you to redirect you _out of the document root_.

So, if you open `http://example.org/install.php`, the InstallationKernel will return the following response header:

```
…
Location: ../contao/install
…
```

This makes no sense of course, since you can't go one level up out of the domain, thus the browser simply opens `contao/install` instead. So

``` php
return new RedirectResponse('../contao/install');
```

simply needs to be changed to

``` php
return new RedirectResponse('contao/install');
```

**Comment by leofeyer** [comment]
> > Looks ok to me. You are supposed to set the document root to the web/ folder.
> 
> It's no a requirement to run Symfony applications though.

I think you are wrong: http://symfony.com/doc/current/cookbook/configuration/web_server_configuration.html

**Comment by fritzmg** [comment]
> I think you are wrong: http://symfony.com/doc/current/cookbook/configuration/web_server_configuration.html

It's not required. It's certainly recommended (as it should be), but it's not required. The Symfony application framework will work without setting the document root to the `/web` folder. The [symfony/symfony-standard-edition](https://github.com/symfony/symfony-standard) even comes with default `.htaccess` files in the [`/app`](https://github.com/symfony/symfony-standard/blob/v3.1.1/app) and [`/src`](https://github.com/symfony/symfony-standard/tree/v3.1.1/src) directory with

``` xml
<IfModule mod_authz_core.c>
    Require all denied
</IfModule>
<IfModule !mod_authz_core.c>
    Order deny,allow
    Deny from all
</IfModule>
```

to prevent any access to these folders on Apache servers by default, in case the application is run without setting the document root to `/web`.

Also the `mod_proxy_cgi` part of the documentation even says

```
# If you run your Symfony application on a subpath of your document root …
```

(http://symfony.com/doc/current/cookbook/configuration/web_server_configuration.html#using-mod-proxy-fcgi-with-apache-2-4)

We and a web agency with which we work together use many Symfony applications this way, mostly in cases where the complete URL to the application is not directly visible to the regular user (Facebook Apps/Tabs or other iframe embeddings for example).

Regardless of any of that, it still does not make sense to redirect _out of the document root_, i.e. into the parent folder of the document root by doing

``` php
return new RedirectResponse('../contao/install');
```

As already pointed out, this only works if you set the document root to `/web` since in that case the browser **has to ignore** the `../` part of the redirect response - otherwise the redirect would lead to nowhere. In cases where you didn't set the document root to `/web` this redirect will go to the parent folder of `/web`, which is wrong of course.

**Comment by leofeyer** [comment]
Anyway, its required by Contao. We even have a [listener](https://github.com/contao/core-bundle/blob/master/src/EventListener/InsecureInstallationListener.php) which ensures that the document root is set correctly.

**Comment by fritzmg** [comment]
> Anyway, its required by Contao. We even have a listener which ensures that the document root is set correctly.

Even there you skip the check, if it's running on localhost. This way you can still have a dev environment without having to set the document root to `/web`:

``` php
// Skip the check on localhost
if (in_array($request->getClientIp(), $this->localIps)) {
    return;
}
```

https://github.com/contao/core-bundle/blob/4.2.0/src/EventListener/InsecureInstallationListener.php#L40

Again, with the current redirect response, i.e. `RedirectResponse('../contao/install')`, you are effectively trying to say **"Move out of the document root and then into contao/install**", which of course is not possible.

Consider opening the Install Tool with

```
http://example.org/install.php
```

That script will respond with the follwing redirect response:

```
../contao/install
```

So with `../` you are saying, "move one level up". But what exactly is one level up from `http://example.org/`?

**Comment by fritzmg** [comment]
@ausi showed me, that `../` can alleviate infinite redirects when accessing the InstallationKernel with a trailing slash. For instance, if the redirect response was only `contao/install` and you access the InstallationKernel like this:

```
example.org/install.php/
```

it would cause an infinite redirect loop (`example.org/install.php/contao/install/contao/install/contao/install…`).

With a redirect response of `../contao/install` you can be correctly redirected to the InstallationController when accessing the InstallationKernel in the following ways:

```
example.org/install.php
example.org/install.php/
example.org/web/install.php/
```

Only

```
example.org/web/install.php
```

will not cause a proper redirect. But as @ausi said in IRC - you can't really know where exactly you want to be redirected to. For instance, if you have no rewrite rules in place, you actually need to be redirected to `app.php/contao/install`. But the InstallationKernel cannot know that.

**Comment by ausi** [comment]
AFAIR the `/contao/` part is also configurable.

**Comment by leofeyer** [comment]
Fixed in 66eea6d2cb7040133267327c87984aa6e73d45ec.
